### ISSUE TITLE
How to move axis label to the middle of the axis?

### ISSUE THREAD

**Issue by topvis** [uber/react-vis#542]
The default position of the axis label is at the end of the axis as shown in the screenshot. Is there anyway to set the label to be in a different position, e.g., at where the arrows point to?

Another question about the axis labels... how to change style (font, size etc.) of the axis labels? I tried search the documentation [here ](http://uber.github.io/react-vis/#/documentation/api-reference/axes)but couldn't find how to do it.

![image](https://user-images.githubusercontent.com/5996531/29232122-f91f96fe-7eb7-11e7-9dcd-011ffa56889c.png)

**Comment by mcnuttandrew** [comment]
That functionality doesn't really exist right now! It certainly can and should be part of the Annotation as first class citizens (#494), but i think you could probably get something working today by adding some custom SVG like so:

```javascript
import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

class CustomAxisLabel extends PureComponent {
  render() {
    return (
      <g transform={`translate(${width / 2}, ${height / 2}`}> <text>LABEL NAME </label></g>
    );
  }
}

CustomAxisLabel.displayName = 'CustomAxisLabel';
CustomAxisLabel.requiresSVG = true;
export default CustomAxisLabel;
```

Then embed that into XYPlot as child, like so: 

```javascript
<XYPlot>
<XAxis>
<LineSeries>
<CustomAxisLabel />
</XYPlot>
```

**Comment by topvis** [comment]
@mcnuttandrew thanks for the solution. I can try the workaround.
I think it would be easier if once can do something like:
```
<XAxis title="X Axis Title" titlePosition="middle-under" style={{
  title: {fontSize: '16px}
}}...
```
titlePosition can be selected from the pre-defined positions.
the style of the title can be set in style property.
just my two cents.

**Comment by mcnuttandrew** [comment]
I totally agree and would welcome a PR to that effect! Maybe a more robust api would be like 

```javascript
<XAxis title="X Axis Title" titlePositionX="50%" titlePositionY="-15px" style={{
  title: {fontSize: '16px}
}}/>
```

or something to that effect so that we dont support a bunch of individual keywords?

**Comment by brandonmp** [comment]
for those seeking a for-now solution, here's what I came up w/


```
const CustomAxisLabel = (props: {
    title: string,
    xAxis: boolean,
    // note these next two are passed down from the parent XYPlot/Flexible*XYPlot
    innerWidth: number,
    innerHeight: number
}) => {
    const transform = props.xAxis
        ? `translate(${props.innerWidth / 2}, ${1.2 * props.innerHeight})`
        : `translate(10,${props.innerHeight / 2 +
              props.title.length * 3}) rotate(-90)`;
    return (
        <g
            transform={transform}
        >
            <text>{props.title}</text>
        </g>
    );
};

CustomAxisLabel.displayName = 'CustomAxisLabel';
CustomAxisLabel.requiresSVG = true;



<CustomAxisLabel title={'This is my Y Axis'}/>
<CustomAxisLabel title={'This is my X Axis'} xAxis />
```


which looks like
![image](https://user-images.githubusercontent.com/14139991/30784863-32fa0248-a12b-11e7-83b6-80ba3ed8a23e.png)

**Comment by mcnuttandrew** [comment]
There's a solution now available in ChartLabel (see #1038 for more details)

**Comment by vtarelkin-comprehend** [comment]
Is it only me - but here is EXACT link how to make this possible
https://github.com/uber/react-vis/commit/dd0ff5efcaa749ff22649b864fdafa6347e564eb
